src/algebraic_topology/dold_kan/degeneracies.lean

```diff
@@ -0,0 +1,144 @@
+/-
+Copyright (c) 2022 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import algebraic_topology.dold_kan.decomposition
+import algebraic_topology.split_simplicial_object
+
+/-!
+
+# Behaviour of P_infty with respect to degeneracies
+
+For any `X : simplicial_object C` where `C` is an abelian category,
+the projector `P_infty : K[X] ⟶ K[X]` is supposed to be the projection
+on the normalized subcomplex, parallel to the degenerate subcomplex, i.e.
+the subcomplex generated by the images of all `X.σ i`.
+
+In this file, we obtain `degeneracy_comp_P_infty` which states that
+if `X : simplicial_object C` with `C` a preadditive category,
+`θ : [n] ⟶ Δ'` is a non injective map in `simplex_category`, then
+`X.map θ.op ≫ P_infty.f n = 0`. It follows from the more precise
+statement vanishing statement `σ_comp_P_eq_zero` for the `P q`.
+
+-/
+
+open category_theory category_theory.category category_theory.limits
+  category_theory.preadditive opposite
+open_locale simplicial dold_kan
+
+namespace algebraic_topology
+
+namespace dold_kan
+
+variables {C : Type*} [category C] [preadditive C]
+
+lemma higher_faces_vanish.comp_σ {Y : C} {X : simplicial_object C} {n b q : ℕ} {φ : Y ⟶ X _[n+1]}
+  (v : higher_faces_vanish q φ) (hnbq : n + 1 = b + q) :
+    higher_faces_vanish q (φ ≫ X.σ ⟨b,
+    by simpa only [hnbq, nat.lt_succ_iff, le_add_iff_nonneg_right] using zero_le q⟩) :=
+λ j hj, begin
+  rw [assoc, simplicial_object.δ_comp_σ_of_gt', fin.pred_succ,
+    v.comp_δ_eq_zero_assoc _ _ hj, zero_comp],
+  { intro hj',
+    simpa only [hj', hnbq, fin.coe_zero, zero_add, add_comm b, add_assoc, false_and,
+      add_le_iff_nonpos_right, le_zero_iff, add_eq_zero_iff, nat.one_ne_zero] using hj, },
+  { simp only [fin.lt_iff_coe_lt_coe, nat.lt_iff_add_one_le,
+      fin.succ_mk, fin.coe_mk, fin.coe_succ, add_le_add_iff_right],
+    linarith, },
+end
+
+lemma σ_comp_P_eq_zero (X : simplicial_object C)
+  {n q : ℕ} (i : fin (n + 1)) (hi : n + 1 ≤ i + q) : (X.σ i) ≫ (P q).f (n + 1) = 0 :=
+begin
+  revert i hi,
+  induction q with q hq,
+  { intros i hi,
+    exfalso,
+    have h := fin.is_lt i,
+    linarith, },
+  { intros i hi,
+    by_cases n+1 ≤ (i : ℕ) + q,
```

Comment:
After the above suggestion, you can remove the intros lines
```suggestion
  { exfalso,
    have h := fin.is_lt i,
    linarith, },
  { by_cases n+1 ≤ (i : ℕ) + q,
```

Reply:
Thanks very much for the suggestion, which I have just done.